### PR TITLE
implement dynamodb streams

### PR DESCRIPTION
**Pull request by kyleaa**
Contains an implementation of the DynamoDB Streams API based on http://docs.aws.amazon.com/dynamodbstreams/latest/APIReference/API_Operations.html. There are integration tests included that run against local dynamodb.

### CODE REVIEW

File: lib/ex_aws/dynamo_streams.ex [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  @type stream_arn :: binary
+
+  @doc """
+  Lists all of the streams associated with an account.
+  If you have multiple streams on a table (created by disabling/enabling the stream)
+  you may have difficulty identifying the active stream from this operation.
+  """
+  @spec list_streams() :: ExAws.Operation.JSON.t
+  def list_streams() do
```
Comment (benwilson512) [inline lib/ex_aws/dynamo_streams.ex]: The optional parameters should be supported from: http://docs.aws.amazon.com/dynamodbstreams/latest/APIReference/API_ListStreams.html

**Comment by kyleaa** [comment]
Added! Thanks!

**Comment by benwilson512** [comment]
Good stuff, thank you!

**Comment by benwilson512** [comment]
Hey so, did you try actually running this against AWS? Because based on the testing I'm doing it's got signature errors.

**Comment by benwilson512** [comment]
This has been reverted as I couldn't easily make it work. I confess I'm a bit annoyed that this was submitted without ever having tested it against the actual AWS service. Merging it to master in that state is however my fault. Please include some tests that actually hit AWS like `list_streams` and when those pass we can try merging this again.

**Comment by kyleaa** [comment]
I sincerely apologize - I am upset with myself as well for this oversight. Thank you for your patience and the time you put into this project!

**Pull request merged**